### PR TITLE
refactor detection predictor

### PR DESCRIPTION
**Pull request by tuofeilunhifi**
# Motivation
The use of unified DetectionPredictor for detection test and predictor.

For example:
1. for detection test
```
        model_path = '.pth'
        config_path = '.py'
        img = './demo.jpg'
        out_file = './result.jpg'
        detector = DetectionPredictor(model_path, config_path)
        output = detector.predict(img)
        detector.visualize(img, output, out_file=out_file)
```
2. for detection predictor
```
        model_path = '.pth'
        # config_path = '.py'   # remove config_path   
        img = './demo.jpg'
        out_file = './result.jpg'
        detector = DetectionPredictor(model_path)
        output = detector.predict(img)
        detector.visualize(img, output, out_file=out_file)
```

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -51,6 +51,33 @@ Please refer to the installation section in [quick_start.md](docs/source/quick_s
 
 ## Get Started
 
+To immediately use a model on a given input image, we provide the Predictor API. Predictor group together a pretrained model with the preprocessing that was used during that model's training. For example, we can easily extract detected objects in an image:
```
Comment (wenmengzhou) [inline README.md:54]: add this paragrath to detection tutorial

File: easycv/predictors/detector.py [hunk 2]
```diff
@@ -36,6 +35,184 @@
     from easycv.thirdparty.mtcnn import FaceDetector
 
 
+@PREDICTORS.register_module()
+class DetectionPredictor(PredictorInterface):
```
Comment (wenmengzhou) [inline easycv/predictors/detector.py:39]: by calling it detection predictor， does it support all the detection models?

File: easycv/predictors/detector.py [hunk 3]
```diff
@@ -36,6 +35,184 @@
...
+
+    def __init__(self, model_path, config_path=None):
+
+        self.model_path = model_path
+
+        if config_path is not None:
+            self.cfg = mmcv_config_fromfile(config_path)
+        else:
+            if is_url_path(self.model_path) and url_path_exists(
```
Comment (wenmengzhou) [inline easycv/predictors/detector.py:48]: how to determine whether the model_path is a checkpoint path or an exported model path?

File: easycv/predictors/detector.py (hunk 2 above)
Comment (tuofeilunhifi) [reply easycv/predictors/detector.py:39]: In addition to yolox, xinyi is still being reconstructed and can be used in a unified manner after the reconstruction is complete

File: easycv/predictors/detector.py (hunk 3 above)
Comment (tuofeilunhifi) [reply easycv/predictors/detector.py:48]: universal. config_path is to avoid inconsistent model_path versions.

File: README.md (hunk 1 above)
Comment (tuofeilunhifi) [reply README.md]: done

File: docs/source/quick_start.md [hunk 4]
```diff
@@ -91,6 +91,35 @@
     You can also verify your installation using following quick-start examples
 
 
+## Get Started
+
+To immediately use a model on a given input image, we provide the Predictor API. Predictor group together a pretrained model with the preprocessing that was used during that model's training. For example, we can easily extract detected objects in an image:
```
Comment (wenmengzhou) [inline docs/source/quick_start.md:96]: move this to detection tutorial

File: easycv/predictors/detector.py [hunk 5]
```diff
@@ -38,6 +38,190 @@
     from easycv.thirdparty.mtcnn import FaceDetector
 
 
+@PREDICTORS.register_module()
+class DetectorPredictor(PredictorInterface):
```
Comment (wenmengzhou) [inline easycv/predictors/detector.py:42]: it does not support all the detectors, rename it  for this time and give it back when it support all the detection moduels

Comment (tuofeilunhifi) [reply easycv/predictors/detector.py]: done

File: docs/source/quick_start.md (hunk 4 above)
Comment (tuofeilunhifi) [reply docs/source/quick_start.md]: done

**Review by wenmengzhou — APPROVED** [review]

**Pull request merged**